doc/dualtor/active_active_hld.md

```diff
@@ -0,0 +1,316 @@
+
+* Data Plane  
+  * Traffic to the server  
+    * Traffic lands on any of the T1 by ECMP from T2s.
+    * T1 forwards traffic to either of the T0s by ECMP. 
+    * T0 sends the traffic to the server and NIC delivers traffic up the stack.
+  * Traffic from the server to outside the cluster  
+    * NIC determines which link to use and sends all the packets on a flow using the same link.
+    * T0 sends the traffic to the T1 by ECMP. 
+  * Traffic from the server to within the cluster  
+    * NIC determines which link to use and sends all the packet on a flow using the same link.
+    * T0 sends the traffic to destination server if T0 has learnt the MAC address of the destination server.
+
+#### 3.1.2 Server Uplink Issue  
+Both T0s are up and functioning and some servers NIC are only connected to 1 ToR (due to cable issue, or the cable is taken out for maintenance).  
+* Control Plane  
+No change from the normal case. 
+* Data Plane  
+  * Traffic to the server  
+    * Traffic lands on any of the T1 by ECMP from T2s.
+    * T1 forwards traffic to either of the T0s by ECMP.
+    * **If T0 does not have the downlink to the server, T0 will send the traffic to the peer T0 over IPinIP encap via T1s.**  
+    * T0 sends the traffic to the server and NIC delivers traffic up the stack. 
+  * Traffic from the server to outside the cluster   
+    * T0 will signal to NIC which side to use.  
+    * NIC determines which link to use and sends all the packets on a flow using the same link. If server NIC has only 1 connection up, all traffic will be on this connection. 
+    * T0 sends the traffic to the T1 by ECMP
+  * Traffic from the server to within the cluster  
+    * T0 will signal to NIC which side to use. 
+    * NIC determines which link to use and sends all the packets on a flow using the same link. If Server NIC has only 1 connection up, all traffic will be on this connection
+    * If T0 does not have the downlink to the server, T0 will send the traffic to the peer T0 over IPinIP encap via T1s. 
+    * T0 sends the traffic to the server.
+
+#### 3.1.3 ToR Failure  
+Only 1 T0s is up and functioning.
+* Control Plane  
+Only 1 T0 will advertise the VLAN (IPv4 and v6) to upstream T1s. 
+* Data Plane  
+  * Traffic to the server  
+    * Traffic lands on any of the T1 by ECMP from T2s.
+    * T1 forwards traffic to either of the T0s by ECMP. If one T0 is down, T1 forwards traffic to the healthy one. 
+    * T0 sends the traffic to the server. 
+  * Traffic from the server to outside the cluster   
+    * T0 will signal to NIC which side to use.  
+    * T0 sends the traffic to the T1 by ECMP.
+  * Traffic from the server to within the cluster  
+    * T0 will signal to NIC which side to use.
+    * T0 sends the traffic to the server. 
+
+#### 3.1.4 Comparison to Active-Standby  
+Highlight on the common and differences with Active-Standby:   
+  ![comparison](./image/difference.png)  
```

Comment:
markdown support table, can we use markdown table format easier to change in the future.

Reply:
Updated.